src/rgw/rgw_user.cc

```diff
@@ -95,6 +95,47 @@ int rgw_user_sync_all_stats(RGWRados *store, const rgw_user& user_id)
 }
 
 /**
+ * Add group name to user info
+ */
+bool rgw_add_group_to_user(RGWUserInfo& info, string& group_name)
```

Comment:
as below, `group_name` should probably be a const ref here, and copied into `info.groups` instead of moved